biodatasets/biorelex/biorelex.py

```diff
@@ -0,0 +1,377 @@
+        relations_ = self._get_relations(example_id, ref_id_map, example["interactions"])
+
+        document_ = {
+            "id": example_id,
+            "document_id": example["paperid"],
+            "passages": [
+                {
+                    "id": example_id + ".sent",
+                    "type": "sentence",
+                    "text": [example["text"]],
+                    "offsets": [[0, len(example["text"])]],
+                }
+            ],
+            "entities": entities_,
+            "coreferences": corefs_,
+            "relations": relations_,
+            "events": [],
+        }
+        return document_
+
+    def _get_entities(self, example_id, entities):
+        entities_ = []
+        corefs_ = []
+
+        eid = it.count(0)
+        cid = it.count(0)
+        # dictionary mapping the original ref ids (indexes of entities) for relations
+        org_rel_ref_id_2_kb_entity_id = defaultdict(list)
+
+        for relation_ref_id, entity in enumerate(entities):
+
+            # get normalization for entities
+            normalized_ = self._get_normalizations(entity)
+
+            # create entity for each synonym
+            coref_eids_ = []
+            for names in entity["names"]:
+                for id, mention in enumerate(names["mentions"]):
+                    entity_id = example_id + ".ent" + str(next(eid)) + "_" + str(id)
+                    org_rel_ref_id_2_kb_entity_id[relation_ref_id].append(entity_id)
+                    coref_eids_.append(entity_id)
+                    entities_.append(
+                        {
+                            "id": entity_id,
+                            "type": entity["label"],
+                            "text": [names["text"]],
+                            "offsets": [mention],
+                            "normalized": normalized_,
+                        }
+                    )
+
+            # create coreferences
+            coref_id = example_id + ".coref" + str(next(cid))
+            corefs_.append(
+                {
+                    "id": coref_id,
+                    "entity_ids": coref_eids_,
+                }
+            )
+        return entities_, corefs_, org_rel_ref_id_2_kb_entity_id
+
+    def _get_normalizations(self, entity):
+        normalized_ = []
+        if entity["grounding"]:
+            assert len(entity["grounding"]) == 1
+            if entity["grounding"][0]["entrez_gene"] != "NA":
+                normalized_.append({"db_name": "entrez", "db_id": entity["grounding"][0]["entrez_gene"]})
+            if entity["grounding"][0]["hgnc_symbol"] != "NA":
+                normalized_.append({"db_name": "hgnc", "db_id": entity["grounding"][0]["hgnc_symbol"]})
+
+            # maybe parse some other ids?
+            source = entity["grounding"][0]["source"]
+            if (
+                source != "NCBI gene" and source != "https://www.genenames.org/data/genegroup/"
+            ):  # NCBI gene is same as entrez
```

Comment:
Exactly: my understanding is that Entrez Gene is the old name of NCBI Gene. Would it be much trouble using `NCBI Gene` in `db_name`?

Reply:
Sure i've changed the name of Entrez to NCBI Gene!